test/builder/pipeline.go

```diff
@@ -399,10 +410,13 @@ func PipelineRunCompletionTime(t time.Time) PipelineRunStatusOp {
 	}
 }
 
-// PipelineRunTaskRunsStatus sets the TaskRuns of the PipelineRunStatus.
-func PipelineRunTaskRunsStatus(taskRuns map[string]*v1alpha1.PipelineRunTaskRunStatus) PipelineRunStatusOp {
+// PipelineRunTaskRunsStatus sets the status of TaskRun to the PipelineRunStatus.
+func PipelineRunTaskRunsStatus(taskRunName string, status *v1alpha1.PipelineRunTaskRunStatus) PipelineRunStatusOp {
 	return func(s *v1alpha1.PipelineRunStatus) {
-		s.TaskRuns = taskRuns
+		if s.TaskRuns == nil {
+			s.TaskRuns = make(map[string]*v1alpha1.PipelineRunTaskRunStatus)
+		}
+		s.TaskRuns[taskRunName] = status
 	}
 }
 
```

Comment:
So, looks like there is already a `Condition` type in the builder package: https://github.com/tektoncd/pipeline/blob/master/test/builder/task.go#L315

I'm thinking we can rename it to `StatusCondition` or `TaskRunStatusCondition` and call this one Condition? (Will probably do that in a different PR since we'd be renaming a bunch of functions)

Reply:
Opened #1127